modin/core/dataframe/pandas/dataframe/dataframe.py

```diff
@@ -192,6 +192,10 @@ def __init__(
                     sum(row_lengths), len(self._index_cache)
                 ),
             )
+            ErrorMessage.catch_bugs_and_request_email(
```

Comment:
We have respective checks in `_filter_empties`. Shouldn't those be enough? We could change `!=` to `< 0` there.

Reply:
Thanks for pointing this out. I think we should keep `_filter_empties` the way it is now, since its purpose is to filter partitions that are empty and don't need computation. Partitions with negative length/width is not excess computation, but an error.

Also, consider if `self._row_lengths = [32, 32, -5]` (it theoretically shouldn't be after this bugfix) and `len(self._index_cache) = 59`. If we use `_filter_empties` to make `self._row_lengths = [32, 32, 0]`, then `sum(self._row_lengths) != len(self._index_cache)` which violates the other `ErrorMessage. catch_bugs_and_request_email` statement.

Reply:
Should we compute `self._row_lengths` to have `[32, 27]` value for in such a case?

Reply:
Yup! The line `else max(0, index_len - (index_chunksize * i))` in https://github.com/modin-project/modin/pull/4368/files#diff-67adc6927585a7b7d0ceebaff3f709c7ad88e80b57e04495924ae9bf4e852b51R138 does this.

In this case, it'll be `59 - (32 * 1) = 27` for `i = 1` and `0` for all `i > 1` (for the remaining partitions).

Reply:
This is O(number of partitions), at the high end there are 10's (maybe 100) thousands of partitions, so this will generally not be too computationally intensive.

Reply:
What if there is a million partitions? If we really need these checks, I wonder if it is beneficial to place these checks, as well as computation of `sum(row_lengths)` and `sum(column_widths)` in for loop to check then them against `len(self._index_cache)` and `len(self._columns_cache)`?

Reply:
ok